lib/pages/login/login_page.dart

```diff
@@ -0,0 +1,103 @@
+        child: Container(
+          decoration: BoxDecoration(
+            image: DecorationImage(
+              fit: BoxFit.cover,
+              image: AssetImage('assets/images/sign_in_background.png'),
+              colorFilter: ColorFilter.mode(
+                  Colors.black.withOpacity(0.4), BlendMode.overlay),
+            ),
+          ),
+          child: BackdropFilter(
+            filter: ImageFilter.blur(sigmaX: 6, sigmaY: 6),
+            child: Padding(
+              padding: const EdgeInsets.only(left: 24, right: 24),
+              child: Form(
+                key: _loginController.formKey,
+                autovalidateMode: AutovalidateMode.onUserInteraction,
+                child: Column(
+                  mainAxisAlignment: MainAxisAlignment.start,
+                  crossAxisAlignment: CrossAxisAlignment.stretch,
+                  children: [
+                    Container(
+                      margin: const EdgeInsets.only(top: 150.0),
+                      child: SvgPicture.asset('assets/images/logo.svg'),
+                    ),
+                    const SizedBox(height: 100),
+                    TextFormField(
+                      keyboardType: TextInputType.emailAddress,
+                      controller: _loginController.emailController,
+                      decoration: InputDecoration(
+                          labelStyle: TextStyle(color: Colors.white30),
+                          floatingLabelBehavior: FloatingLabelBehavior.never,
+                          border: OutlineInputBorder(
+                              borderSide: BorderSide.none,
+                              borderRadius: BorderRadius.circular(12.0)),
+                          fillColor: Colors.white24,
+                          filled: true,
+                          labelText:
+                              AppLocalizations.of(context).titleGeneralEmail),
+                      style: TextStyle(color: Colors.white),
+                      validator: _emailValidator,
+                    ),
+                    const SizedBox(height: 20),
+                    TextFormField(
+                      keyboardType: TextInputType.text,
+                      controller: _loginController.passwordController,
+                      decoration: InputDecoration(
+                          labelStyle: TextStyle(color: Colors.white30),
+                          floatingLabelBehavior: FloatingLabelBehavior.never,
+                          border: OutlineInputBorder(
+                              borderSide: BorderSide.none,
+                              borderRadius: BorderRadius.circular(12.0)),
+                          fillColor: Colors.white24,
+                          filled: true,
+                          labelText: AppLocalizations.of(context)
+                              .titleGeneralPassword),
+                      obscureText: true,
+                      style: TextStyle(color: Colors.white),
+                    ),
```

Comment:
What do you think about creating a reuseable widget for this TextFormField? It seems to duplicate a lot
```dart
Widget _TextFormField(TextEditingController controller, String labelText, String validator) {
  return TextFormField(
   keyboardType: TextInputType.emailAddress,
   controller: controller,
   decoration: InputDecoration(
       labelStyle: TextStyle(color: Colors.white30),
       floatingLabelBehavior: FloatingLabelBehavior.never,
       border: OutlineInputBorder(
           borderSide: BorderSide.none,
           borderRadius: BorderRadius.circular(12.0)),
       fillColor: Colors.white24,
       filled: true,
       labelText:
           labelText,
   style: TextStyle(color: Colors.white),
   validator: validator,
  )
}
```

Reply:
I felt the same too, at first I plan to have another refactor PR later 🤔

Reply:
keyboardType and textObscue for password field is different from email or a normal text field, @markgravity 🤔  I think it might be a bit minor but we may only or better just extract the decoration parts somehow. Extracting View to a function may work for some special cases only I can foresee 🤔

Reply:
@sleepylee Agree, I think we should extract the decoration would be better there👍